elastic/beats
# Fix reading of root paths for V1 cgroups events

## What does this PR do?

This is a fix for https://github.com/elastic/beats/issues/30977 , Where the new code wasn't properly filtering controller paths before trying to find the "root" controller ID. This also cleans up some of the cgroup testing code, including the testdata files, since it was a pain.

This is currently a draft PR, as it needs to get tested against a suite of different cgroups implementations.

## Why is it important?

This is a bugfix.

## Checklist



## How to test this PR locally
- Build/run on a linux system with V1 or hybrid cgroups
- run the `system/process` metricset with cgroups enabled
- Examine output, make sure `id` fields for processes are populated.
